src/agent/__tests__/MessageSender.test.ts

```diff
@@ -0,0 +1,144 @@
+  transportServiceFindEndpointMock.mockReturnValue(endpoint)
+
+  let messageSender: MessageSender
+  let outboundTransporter: OutboundTransporter
+  let connection: ConnectionRecord
+
+  describe('sendMessage', () => {
+    beforeEach(() => {
+      outboundTransporter = new DummyOutboundTransporter()
+      messageSender = new MessageSender(enveloperService, transportService, logger)
+      connection = getMockConnection({ id: 'test-123' })
+    })
+
+    test('throws error when there is no outbound transport', async () => {
+      const message = new AgentMessage()
+      const outboundMessage = createOutboundMessage(connection, message)
+      await expect(messageSender.sendMessage(outboundMessage)).rejects.toThrow(`Agent has no outbound transporter!`)
+    })
+
+    test('calls transporter with connection, payload and endpoint', async () => {
+      const message = new AgentMessage()
+      const spy = jest.spyOn(outboundTransporter, 'sendMessage')
+      const outboundMessage = createOutboundMessage(connection, message)
+      messageSender.setOutboundTransporter(outboundTransporter)
+
+      await messageSender.sendMessage(outboundMessage)
+
+      const [[sendMessageCall]] = spy.mock.calls
+      expect(sendMessageCall).toEqual({
+        connection,
+        payload: wireMessage,
+        endpoint,
+        responseRequested: false,
+        session,
+      })
+    })
+  })
+
+  describe('packMessage', () => {
+    beforeEach(() => {
+      outboundTransporter = new DummyOutboundTransporter()
+      messageSender = new MessageSender(enveloperService, transportService, logger)
+      connection = getMockConnection({ id: 'test-123' })
+    })
+
+    test('returns outbound message context with connection, payload and endpoint', async () => {
+      const message = new AgentMessage()
+      const outboundMessage = createOutboundMessage(connection, message)
+
+      const result = await messageSender.packMessage(outboundMessage)
+
+      expect(result).toEqual({
+        connection,
+        payload: wireMessage,
+        endpoint,
+        responseRequested: false,
+      })
+    })
+
+    test('when message has return route returns outbound message context with responseRequested', async () => {
+      const message = new AgentMessage()
+      message.setReturnRouting(ReturnRouteTypes.all)
+      const outboundMessage = createOutboundMessage(connection, message)
+
+      const result = await messageSender.packMessage(outboundMessage)
+
+      expect(result.responseRequested).toEqual(true)
+    })
+  })
+})
+
+/**
+ * Returns mock of function with correct type annotations according to original function `fn`.
+ * It can be used also for class methods.
+ *
+ * @param fn function you want to mock
+ * @returns mock function with type annotations
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function mockFunction<T extends (...args: any[]) => any>(fn: T): jest.MockedFunction<T> {
```

Comment:
Can't we import this from the helpers file?

Reply:
Ahh, I didn't know about that. I updated the JS doc at least :)